js/src/common/Translator.ts

```diff
@@ -4,34 +4,44 @@ import username from './helpers/username';
 import extract from './utils/extract';
 
 export default class Translator {
-  constructor() {
-    /**
-     * A map of translation keys to their translated values.
-     *
-     * @type {Object}
-     * @public
-     */
-    this.translations = {};
+  /**
+   * A map of translation keys to their translated values.
+   */
+  translations = {};
+
+  /**
+   * The underlying ICU MessageFormatter util.
+   *
+   * @ts-ignore
+   */
+  protected formatter = new RichMessageFormatter(null, this.formatterTypeHandlers(), mithrilRichHandler);
+
+  setLocale(locale: string) {
+    this.formatter.locale = locale;
+  }
 
-    this.formatter = new RichMessageFormatter(null, this.formatterTypeHandlers(), mithrilRichHandler);
+  addTranslations(translations: { [key: string]: string }) {
```

Comment:
We should use the `Record` type for this.

```suggestion
  addTranslations(translations: Record<string, string>) {
```